src/Scientist/Internals/ExperimentInstance.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace GitHub.Internals
+{
+    /// <summary>
+    /// An instance of an experiment. This actually runs the control and the candidate and measures the result.
+    /// </summary>
+    /// <typeparam name="T">The return type of the experiment</typeparam>
+    internal class ExperimentInstance<T>
+    {
+        static Random _random = new Random(DateTimeOffset.UtcNow.Millisecond);
+        
+        readonly Dictionary<string, Func<Task<T>>> _behaviors;
+        readonly string _name;
+
+        public ExperimentInstance(string name, Func<T> control, Func<T> candidate)
+            : this(name, () => Task.FromResult(control()), () => Task.FromResult(candidate()))
+        {
+        }
+
+        public ExperimentInstance(string name, Func<Task<T>> control, Func<Task<T>> candidate)
+        {
+            _name = name;
+            _behaviors = new Dictionary<string, Func<Task<T>>>
+            {
+                { "control", control },
```

Comment:
This is to follow up with a change to allow multiple `Try` calls.  Alternately, it could be a `List` that has a property for storing the Name and Delegate.  What do you think?